### PR TITLE
fix(editor): Ensure default credential values are not detected as dirty state

### PR DESCRIPTION
**Pull request by ivov**
To verify, go to the credential creation modal for a cred having default values (e.g. RabbitMQ) and close the modal. There should be no prompt asking about unsaved changes.

### CODE REVIEW

File: packages/editor-ui/src/components/CredentialEdit/CredentialEdit.vue [hunk 1]
```diff
@@ -670,6 +670,9 @@ export default defineComponent({
 		},
 		// eslint-disable-next-line @typescript-eslint/no-explicit-any
 		onDataChange({ name, value }: { name: string; value: any }) {
+			// disregard default value being set on component mount
```
Comment (OlegIvaniv) [inline packages/editor-ui/src/components/CredentialEdit/CredentialEdit.vue:673]: Super nitpicking 😛, but the current comment is a bit confusing as the method doesn't know if it's being run on mount or in a different context. IMO, a more fitting comment could be `// Skip update if new value matches the current`

**Review by OlegIvaniv — DISMISSED** [review]
Works as expected, left just one nitpicky comment but feel free to merge

File: packages/editor-ui/src/components/CredentialEdit/CredentialEdit.vue (hunk 1 above)
Comment (ivov) [reply packages/editor-ui/src/components/CredentialEdit/CredentialEdit.vue]: Totally agree, updated!

**Review by OlegIvaniv — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.0.4`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.0.4)

**Pull request merged**